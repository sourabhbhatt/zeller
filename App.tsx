import React, {memo} from 'react';
import {ApolloProvider} from '@apollo/client';
import {
  StatusBar,
  SafeAreaView,
  StyleSheet,
  View,
  Platform,
} from 'react-native';
import {SafeAreaProvider} from 'react-native-safe-area-context';

import Colors from './src/assets/colors';
import client from './src/graphql/apolloClient';
import AppNavigator from './src/navigation/AppNavigator';
import ErrorBoundary from './src/components/ErrorBoundary';

const App = () => {
  return (
    <View style={styles.container} testID="app-root">
      <ErrorBoundary ENV="dev">
        <ApolloProvider client={client}>
          <SafeAreaProvider>
            <SafeAreaView style={styles.safeAreaTop} />
            <View style={styles.container} testID="app-container">
              <StatusBar
                barStyle="light-content"
                backgroundColor={Colors.headerAndStatusBar || 'transparent'}
                translucent
              />
              <AppNavigator />
            </View>
          </SafeAreaProvider>
        </ApolloProvider>
      </ErrorBoundary>
    </View>
  );
};

export default memo(App);

const styles = StyleSheet.create({
  safeAreaTop: {
    flex: 0,
    backgroundColor: Colors.headerAndStatusBar || 'transparent',
    paddingTop: Platform.OS === 'android' ? StatusBar.currentHeight || 0 : 0,
  },
  container: {
    flex: 1,
    backgroundColor: Colors.background || '#fff',
  },
});
