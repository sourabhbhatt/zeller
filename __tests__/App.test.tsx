import React from 'react';
import {render} from '@testing-library/react-native';
import App from '../App';

jest.mock('../src/graphql/apolloClient', () => ({
  default: {
    query: jest.fn(),
    mutate: jest.fn(),
  },
}));

jest.mock('../src/navigation/AppNavigator', () => jest.fn(() => <></>));

jest.mock('../src/components/ErrorBoundary', () =>
  jest.fn(({children}) => children),
);

describe('App Component', () => {
  it('renders without crashing', () => {
    const {toJSON} = render(<App />);
    expect(toJSON()).toMatchSnapshot();
  });
});
