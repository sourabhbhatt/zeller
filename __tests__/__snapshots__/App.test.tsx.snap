// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component renders without crashing 1`] = `
<View
  style={
    {
      "backgroundColor": "#F5F5F5",
      "flex": 1,
    }
  }
  testID="app-root"
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      [
        {
          "flex": 1,
        },
        undefined,
      ]
    }
  />
</View>
`;
